File: leetcode/C++/valid-palindrome-ii.cpp
class Solution {
public:
    bool validPalindrome(string s) {
        int i=0, j=(int)s.length()-1,a=0,b=0;
        for(; i<j; ++i,--j) if(s[i]!=s[j]) break;
        for(a=i+1,b=j; a<b; ++a,--b) if(s[a]!=s[b]) break;
        if(a>=b) return true;
        for(a=i,b=j-1; a<b; ++a,--b) if(s[a]!=s[b]) break;
        return a>=b;
    }
};

File: leetcode/Python/valid-palindrome-ii.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        i, j, a, b = 0, len(s)-1, 0, 0
        while i<j:
            if s[i]!=s[j]:
                break
            i, j = i + 1 , j - 1
        a, b = i + 1 , j
        while a<b:
            if s[a]!=s[b]:
                break
            a, b = a + 1, b - 1
        if a>=b:
            return True
        a, b = i, j - 1
        while a<b:
            if s[a]!=s[b]:
                break
            a, b = a + 1, b - 1
        return a>=b